textattack/goal_functions/classification/input_reduction.py

```diff
@@ -0,0 +1,36 @@
+from .classification_goal_function import ClassificationGoalFunction
+
+
+class InputReduction(ClassificationGoalFunction):
+    """
+    An targeted attack on classification models which attempts to maximize the 
+    score of the target label until it is the predicted label.
+    """
+
+    def __init__(self, *args, target_num_words=1, **kwargs):
+        self.target_num_words = target_num_words
+        super().__init__(*args, **kwargs)
+
+    def _is_goal_complete(self, model_output, attacked_text):
+        return (
+            self.ground_truth_output == model_output.argmax()
+            and attacked_text.num_words <= target_num_words
+        )
+
+    def _should_skip(self, model_output, attacked_text):
+        return self.ground_truth_output != model_output.argmax()
+
+    def _get_score(self, model_output, attacked_text):
+        if self.ground_truth_output != model_output.argmax():
+            return float("-inf")
```

Comment:
why do this here and not in untargeted/targeted goal functions? shouldn't this be skipped anyway?

Reply:
Well the goal of input reduction is to make sure the class score isn't changed. It won't be skipped if the class score is changed through the process of removing words. This isn't in untargeted/targeted goal functions because there isn't the same sort of "maximize X conditioned on Y" sort of property that input reduction has ("maximize words removed conditioned on the predicted class not changing")